script/lib/application_insights.py

```diff
@@ -0,0 +1,60 @@
+import os
+import sys
+import json
+import requests
+
+
+def ai_request(query):
+    APP_INSIGHTS_APPLICATION_GUID = os.environ["APP_INSIGHTS_APPLICATION_GUID"]
+    APP_INSIGHTS_API_KEY = os.environ["APP_INSIGHTS_API_KEY"]
+
+    params = {"query": query}
+    headers = {"X-Api-Key": APP_INSIGHTS_API_KEY}
+
+    url = f"https://api.applicationinsights.io/v1/apps/{APP_INSIGHTS_APPLICATION_GUID}/query"
+
+    response = requests.get(url, headers=headers, params=params)
+    return json.loads(response.text)
+
+
+def single_domain_event(event_type):
+
+    # Grab the custom events for a single event type
+    query = f"""customEvents
+         | where cloud_RoleName in ('hmpps-domain-event-logger')
+         | where name == '{event_type}'
+         | where timestamp between((ago(7d)) .. now())
+    """
+
+    full_response = ai_request(query)
+
+    # Do we have any messages?
+    if len(full_response["tables"][0]["rows"]) == 0:
+        sys.exit(
+            f"No domain event message found for '{event_type}' in Application Insights in the previous three days"
```

Comment:
Minor thing - should this be seven days as well?

Reply:
Thanks - fixed